### PR TITLE
(feat): Consistent  header for CardEditor 

### LINKED ISSUE #1561: [CardGalleryHeader] dashboard builder/editor pane header
### What package is this for?

- [x] React
- [ ] Angular

### Summary

In support of the dashboard builder/editor #1372

The Gallery/Editing pane should have a standardized header. From the spec it appears to have two states/configurations
1. Gallery
1. Return to Gallery

This would be a very small component but something that we could publish for ease of design consistency and standardization. Would need to accept props: 
- i18n object prop for the strings
- a `bool` prop `isGallery` which would toggle the two different states. 
- callback for onGalleryClose (the arrow)
- callback for onReturnToGallery (the "open gallery" link and the icon)

**Additional context**

image
image

### PR DESCRIPTION
**Pull request by davidicus**
Closes #1561

**Summary**

- This PR uses a consistently styled header that is part of the `CardEditor` component for both the "Gallery" section and the "Add a card" section. We no longer utilize `SimpleList` header

**Change List (commits, features, bugs, etc)**

- Added the "gallery" header section to `CardEditor.jsx`
- change `title` prop for `SimpleList` in `CardGalleryList.jsx`
- add styles to `_card-editor.scss`

**Acceptance Test (how to verify the PR)**

- Go to Dashboard editor [story](https://deploy-preview-1870--carbon-addons-iot-react.netlify.app/)
- Click on card type to add. 
- Click on "Add a card" header
- see that styles look the same as production with the change of making the "add to card" header have the same height as the "Gallery" header.

### CODE REVIEW

File: src/components/CardEditor/_card-editor.scss [hunk 1]
```diff
@@ -17,6 +17,7 @@
     }
     button.gallery-button {
       /* make ghost button stretch full width */
+      height: calc(100% - 0.5rem);
```
Comment (bjornalm) [inline src/components/CardEditor/_card-editor.scss:20]: what is the 0.5rem for? We could use a sass variable to explain it

**Review by bjornalm — APPROVED** [review]
Nice work

**Review by tay1orjones — APPROVED** [review]

**Pull request merged**